Trie.Keys() panic

### desc
new trie object, and call 'Keys'

### panic code

```
trie.New().Keys()
```

### panic stack

```
panic: runtime error: makeslice: cap out of range

goroutine 1 [running]:
github.com/derekparker/trie.collect(0xc00007e2a0, 0xc0000dde58, 0x0, 0x20)
	/Users/fmt/work/gopath1/src/github.com/derekparker/trie/trie.go:245 +0xad
github.com/derekparker/trie.Trie.PrefixSearch(0x0, 0xc00007e2a0, 0x0, 0x0, 0x0, 0x0, 0x4, 0xc00000e360)
	/Users/fmt/work/gopath1/src/github.com/derekparker/trie/trie.go:143 +0xaa
github.com/derekparker/trie.(*Trie).Keys(...)
	/Users/fmt/work/gopath1/src/github.com/derekparker/trie/trie.go:126

```

--- a/trie.go
+++ b/trie.go
@@ -123,6 +123,10 @@ func (t *Trie) Remove(key string) {
 
 // Returns all the keys currently stored in the trie.
 func (t *Trie) Keys() []string {
+	if t.size == 0 {
+		return []string{}
+	}
+
 	return t.PrefixSearch("")
 }
 

--- a/trie_test.go
+++ b/trie_test.go
@@ -129,25 +129,34 @@ func TestRemove(t *testing.T) {
 }
 
 func TestTrieKeys(t *testing.T) {
-	trie := New()
-	expected := []string{"bar", "foo"}
-
-	for _, key := range expected {
-		trie.Add(key, nil)
+	tableTests := []struct {
+		name         string
+		expectedKeys []string
+	}{
+		{"Two", []string{"bar", "foo"}},
+		{"One", []string{"foo"}},
+		{"Empty", []string{}},
 	}
 
-	kl := len(trie.Keys())
-	if kl != 2 {
-		t.Errorf("Expected 2 keys, got %d, keys were: %v", kl, trie.Keys())
-	}
+	for _, test := range tableTests {
+		t.Run(test.name, func(t *testing.T) {
+			trie := New()
+			for _, key := range test.expectedKeys {
+				trie.Add(key, nil)
+			}
 
-	keys := trie.Keys()
+			keys := trie.Keys()
+			if len(keys) != len(test.expectedKeys) {
+				t.Errorf("Expected %v keys, got %d, keys were: %v", len(test.expectedKeys), len(keys), trie.Keys())
+			}
 
-	sort.Strings(keys)
-	for i, key := range keys {
-		if key != expected[i] {
-			t.Errorf("Expected %#v, got %#v", expected[i], key)
-		}
+			sort.Strings(keys)
+			for i, key := range keys {
+				if key != test.expectedKeys[i] {
+					t.Errorf("Expected %#v, got %#v", test.expectedKeys[i], key)
+				}
+			}
+		})
 	}
 }
 

